infrastructure-petstore-rest-client/src/test/java/today/stepbeyond/examples/springbootexamples/infrastructure/gateways/JsonDifferenceAssertion.java

```diff
@@ -0,0 +1,46 @@
+package today.stepbeyond.examples.springbootexamples.infrastructure.gateways;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.Map;
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.Assertions;
+import wiremock.com.google.common.collect.MapDifference;
+import wiremock.com.google.common.collect.Maps;
+
+public class JsonDifferenceAssertion extends AbstractAssert<JsonDifferenceAssertion, String> {
+
+  protected JsonDifferenceAssertion(String s) {
+    super(s, JsonDifferenceAssertion.class);
+  }
+
+  public static JsonDifferenceAssertion assertThat(String actual) {
+    return new JsonDifferenceAssertion(actual);
+  }
+
+  public void isSameAs(String expected) {
```

Comment:
Nit: I think that this should be named isEqualTo instead. From my understanding, assertJs conventions differentiates them like this: 
- isSameAs: checks if 2 objects are same (that they point to the same reference)
- isEqualTo: checks if objects contents are the same

Reply:
Good point. I felt not good about isEqualTo first, as the strings can actually be different. But we focus on Jsons in this class, so isEqualTo is more precise.